{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": "Seq('CTACTCAGCCCCAGCGGAGGTGAAGGACGTCCTTCCCCAGGAGCCGGTGAGAAG...GCA')"
     },
     "execution_count": 149,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "from Bio.Seq import MutableSeq\n",
    "\n",
    "read = SeqIO.read('databases/APOE2.fna', 'fasta')\n",
    "read.seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "outputs": [
    {
     "data": {
      "text/plain": "Seq('CTACTCAGCCCCAGCGGAGG')"
     },
     "execution_count": 150,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "seq_sl = read.seq[0:20]\n",
    "seq_sl"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "outputs": [
    {
     "data": {
      "text/plain": "MutableSeq('LLSPSG')"
     },
     "execution_count": 151,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Original\n",
    "mutable_seq = MutableSeq(seq_sl)\n",
    "mem = mutable_seq.transcribe().translate()\n",
    "mem"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "#### Silent Mutation"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "outputs": [
    {
     "data": {
      "text/plain": "MutableSeq('TTACTCAGCCCCAGCGGAGG')"
     },
     "execution_count": 152,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mutable_seq[0] = \"T\"\n",
    "mutable_seq"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Original:\t LLSPSG\n",
      "Silent:\t\t LLSPSG\n"
     ]
    }
   ],
   "source": [
    "mem2 = mutable_seq.transcribe().translate()\n",
    "print('Original:\\t', mem)\n",
    "print('Silent:\\t\\t', mem2)"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "Изменили один нуклеотид, мутация прошла не замеченной (аминокислотная последовательность белка осталась неизменной)."
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%% md\n"
    }
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "#### Nonsense"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%% md\n"
    }
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "Для получения Nonsense мутации мы должны получить один из трех стоп-кодонов: UAA, UAG, UGA."
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%% md\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "outputs": [
    {
     "data": {
      "text/plain": "MutableSeq('CTACTCAGCCCCAGCGGAGG')"
     },
     "execution_count": 154,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mutable_seq = MutableSeq(seq_sl)\n",
    "mutable_seq"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "outputs": [
    {
     "data": {
      "text/plain": "MutableSeq('CTATAAAGCCCCAGCGGAGG')"
     },
     "execution_count": 155,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mutable_seq[3:6] = 'TAA' # convert to UAA\n",
    "mutable_seq"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Original:\t LLSPSG\n",
      "Nonsense:\t L*SPSG\n"
     ]
    }
   ],
   "source": [
    "mem3 = mutable_seq.transcribe().translate()\n",
    "print('Original:\\t', mem)\n",
    "print('Nonsense:\\t', mem3)"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "#### Missense"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%% md\n"
    }
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "Должны получить измененную последовательность."
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%% md\n"
    }
   }
  },
  {
   "cell_type": "markdown",
   "source": [],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%% md\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "outputs": [
    {
     "data": {
      "text/plain": "MutableSeq('CTACTCAGCCCCAGCGGAGG')"
     },
     "execution_count": 157,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mutable_seq = MutableSeq(seq_sl)\n",
    "mutable_seq"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "outputs": [
    {
     "data": {
      "text/plain": "MutableSeq('ACACTCAGCCCCAGCGGAGG')"
     },
     "execution_count": 160,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mutable_seq[1] = 'C'\n",
    "mutable_seq"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "outputs": [],
   "source": [
    "mem3 = mutable_seq.transcribe().translate()"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Original:\t LLSPSG\n",
      "Missense:\t TLSPSG\n"
     ]
    }
   ],
   "source": [
    "print('Original:\\t', mem)\n",
    "print('Missense:\\t', mem3)"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "# SNP"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%% md\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "outputs": [],
   "source": [
    "import sys\n",
    "import time\n",
    "from Bio import Entrez"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "outputs": [],
   "source": [
    "Entrez.email = \"omenbestg@gmail.com\" # provide your user email\n",
    "# RECOMMENDED: apply for API key from NCBI (https://ncbiinsights.ncbi.nlm.nih.gov/2017/11/02/new-api-keys-for-the-e-utilities/).\n",
    "# 10 queries per second with a valid API key, otherwise 3 queries per seconds are allowed for 'None'\n",
    "Entrez.api_key = None\n",
    "\n",
    "# dbSNP supported query terms (https://www.ncbi.nlm.nih.gov/snp/docs/entrez_help/) can be build and test online using web query builder (https://www.ncbi.nlm.nih.gov/snp/advanced)\n",
    "# esearch handle\n",
    "eShandle = Entrez.esearch(db=\"snp\",  # search dbSNP\n",
    "                          #complex query for missense and pathogenic variants in LPL gene with global MAF betweeen 0 and 0.01.\n",
    "                          term='348[GeneID] AND pathogenic[Clinical_Significance] AND missense variant[Function_Class] AND (00000.0000[GLOBAL_MAF] : 00000.0100[GLOBAL_MAF])',\n",
    "                          usehistory=\"y\", #cache result on server for download in batches\n",
    "                          retmax=20 # return 20 RSID max\n",
    "                         )"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "outputs": [],
   "source": [
    "# get esearch result\n",
    "eSresult = Entrez.read(eShandle)"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Count : 8\n",
      "RetMax : 8\n",
      "RetStart : 0\n",
      "QueryKey : 1\n",
      "WebEnv : MCID_65b41c0213f1a854cd561b9b\n",
      "IdList : ['201672011', '199768005', '190853081', '140808909', '121918393', '121918392', '769455', '769452']\n",
      "TranslationSet : []\n",
      "TranslationStack : [{'Term': '348[GeneID]', 'Field': 'GeneID', 'Count': '2862', 'Explode': 'N'}, {'Term': 'pathogenic[Clinical_Significance]', 'Field': 'Clinical_Significance', 'Count': '130269', 'Explode': 'N'}, 'AND', {'Term': 'missense variant[Function_Class]', 'Field': 'Function_Class', 'Count': '10256260', 'Explode': 'N'}, 'AND', {'Term': '00000.0000[GLOBAL_MAF]', 'Field': 'GLOBAL_MAF', 'Count': '0', 'Explode': 'N'}, {'Term': '00000.0100[GLOBAL_MAF]', 'Field': 'GLOBAL_MAF', 'Count': '0', 'Explode': 'N'}, 'RANGE', 'GROUP', 'AND']\n",
      "QueryTranslation : 348[GeneID] AND pathogenic[Clinical_Significance] AND missense variant[Function_Class] AND (00000.0000[GLOBAL_MAF] : 00000.0100[GLOBAL_MAF])\n"
     ]
    }
   ],
   "source": [
    "# review results\n",
    "for k in eSresult:\n",
    "    print (k, \":\", eSresult[k])"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "outputs": [],
   "source": [
    "# get result RSIDs list 'Idlist'\n",
    "# total rs count\n",
    "rslist = (eSresult['IdList'])"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "201672011\n",
      "199768005\n",
      "190853081\n",
      "140808909\n",
      "121918393\n",
      "121918392\n",
      "769455\n",
      "769452\n"
     ]
    }
   ],
   "source": [
    "# retmax = 20 so print only 20 RSIDs\n",
    "# additional results can be retrieved by batches\n",
    "# download in batches example http://biopython.org/DIST/docs/tutorial/Tutorial.html#htoc139 or see below.\n",
    "for rs in rslist:\n",
    "    print(rs)"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "outputs": [],
   "source": [
    "# get the WebEnv session cookie, and the QueryKey:\n",
    "\n",
    "webenv = eSresult[\"WebEnv\"]\n",
    "query_key = eSresult[\"QueryKey\"]\n",
    "total_count = int(eSresult[\"Count\"])\n",
    "query_key = eSresult[\"QueryKey\"]\n",
    "retmax = 2 # return 2 rs per batch example"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Going to download record 1 to 2\n",
      "b'<?xml version=\"1.0\" ?>\\n<ExchangeSet xmlns:xsi=\"https://www.w3.org/2001/XMLSchema-instance\" xmlns=\"https://www.ncbi.nlm.nih.gov/SNP/docsum\" xsi:schemaLocation=\"https://www.ncbi.nlm.nih.gov/SNP/docsum ftp://ftp.ncbi.nlm.nih.gov/snp/specs/docsum_eutils.xsd\" ><DocumentSummary uid=\"201672011\"><SNP_ID>201672011</SNP_ID><ALLELE_ORIGIN/><GLOBAL_MAFS><MAF><STUDY>1000Genomes</STUDY><FREQ>A=0.000625/3</FREQ></MAF><MAF><STUDY>ExAC</STUDY><FREQ>A=0.000159/18</FREQ></MAF><MAF><STUDY>GnomAD</STUDY><FREQ>A=0.000492/69</FREQ></MAF><MAF><STUDY>GnomAD_exomes</STUDY><FREQ>A=0.00012/30</FREQ></MAF><MAF><STUDY>PAGE_STUDY</STUDY><FREQ>A=0.001513/119</FREQ></MAF><MAF><STUDY>TOMMO</STUDY><FREQ>A=0.000106/2</FREQ></MAF><MAF><STUDY>TOPMED</STUDY><FREQ>A=0.001024/271</FREQ></MAF><MAF><STUDY>ALFA</STUDY><FREQ>A=0.00016/2</FREQ></MAF></GLOBAL_MAFS><GLOBAL_POPULATION/><GLOBAL_SAMPLESIZE>0</GLOBAL_SAMPLESIZE><SUSPECTED/><CLINICAL_SIGNIFICANCE>likely-benign,pathogenic</CLINICAL_SIGNIFICANCE><GENES><GENE_E><NAME>APOE</NAME><GENE_ID>348</GENE_ID></GENE_E></GENES><ACC>NC_000019.10</ACC><CHR>19</CHR><HANDLE>1000GENOMES,HUMAN_LONGEVITY,1000G_HIGH_COVERAGE,PAGE_CC,TOMMO_GENOMICS,EVA,EVA_EXAC,TOPMED,NCBI-CURATED-RECORDS,ILLUMINA,GNOMAD</HANDLE><SPDI>NC_000019.10:44907806:G:A</SPDI><FXN_CLASS>missense_variant,coding_sequence_variant</FXN_CLASS><VALIDATED>by-frequency,by-alfa,by-cluster</VALIDATED><DOCSUM>HGVS=NC_000019.10:g.44907807G&gt;A,NC_000019.9:g.45411064G&gt;A,NG_007084.2:g.7026G&gt;A,NM_000041.4:c.91G&gt;A,NM_000041.3:c.91G&gt;A,NM_000041.2:c.91G&gt;A,NM_001302688.2:c.169G&gt;A,NM_001302688.1:c.169G&gt;A,NM_001302690.2:c.91G&gt;A,NM_001302690.1:c.91G&gt;A,NM_001302691.2:c.91G&gt;A,NM_001302691.1:c.91G&gt;A,NM_001302689.2:c.91G&gt;A,NM_001302689.1:c.91G&gt;A,NP_000032.1:p.Glu31Lys,NP_001289617.1:p.Glu57Lys,NP_001289619.1:p.Glu31Lys,NP_001289620.1:p.Glu31Lys,NP_001289618.1:p.Glu31Lys|SEQ=[G/A]|LEN=1|GENE=APOE:348</DOCSUM><TAX_ID>9606</TAX_ID><ORIG_BUILD>137</ORIG_BUILD><UPD_BUILD>156</UPD_BUILD><CREATEDATE>2012/05/04 13:09</CREATEDATE><UPDATEDATE>2022/10/13 16:55</UPDATEDATE><SS>489170609,537712914,1363326160,1693665315,1959869648,2095085219,2226012193,2744090381,2750218796,2963330064,3021911540,3625742792,3625742793,3652337639,3652337640,3725733569,3772016988,5076165572,5227950565,5307395571,5435087015,5613191099,5786644758,5847846235,5928256250,5979547880</SS><ALLELE>R</ALLELE><SNP_CLASS>snv</SNP_CLASS><CHRPOS>19:44907807</CHRPOS><CHRPOS_PREV_ASSM>19:45411064</CHRPOS_PREV_ASSM><TEXT/><SNP_ID_SORT>0201672011</SNP_ID_SORT><CLINICAL_SORT>1</CLINICAL_SORT><CITED_SORT/><CHRPOS_SORT>0044907807</CHRPOS_SORT><MERGED_SORT>0</MERGED_SORT></DocumentSummary>\\n<DocumentSummary uid=\"199768005\"><SNP_ID>199768005</SNP_ID><ALLELE_ORIGIN/><GLOBAL_MAFS><MAF><STUDY>1000Genomes</STUDY><FREQ>A=0.000312/2</FREQ></MAF><MAF><STUDY>ExAC</STUDY><FREQ>A=0.00135/25</FREQ></MAF><MAF><STUDY>GnomAD</STUDY><FREQ>A=0.0005/70</FREQ></MAF><MAF><STUDY>GnomAD_exomes</STUDY><FREQ>A=0.000452/75</FREQ></MAF><MAF><STUDY>GoESP</STUDY><FREQ>A=0.000789/10</FREQ></MAF><MAF><STUDY>TOPMED</STUDY><FREQ>A=0.000427/113</FREQ></MAF><MAF><STUDY>ALFA</STUDY><FREQ>A=0.000463/13</FREQ></MAF></GLOBAL_MAFS><GLOBAL_POPULATION/><GLOBAL_SAMPLESIZE>0</GLOBAL_SAMPLESIZE><SUSPECTED/><CLINICAL_SIGNIFICANCE>likely-benign,benign,pathogenic</CLINICAL_SIGNIFICANCE><GENES><GENE_E><NAME>APOE</NAME><GENE_ID>348</GENE_ID></GENE_E></GENES><ACC>NC_000019.10</ACC><CHR>19</CHR><HANDLE>ILLUMINA,HUMAN_LONGEVITY,1000G_HIGH_COVERAGE,EVA_DECODE,EVA-GONL,SWEGEN,EVA,EXOME_CHIP,AFFY,EVA_EXAC,TOPMED,NCBI-CURATED-RECORDS,NHLBI-ESP,GNOMAD</HANDLE><SPDI>NC_000019.10:44909056:T:A</SPDI><FXN_CLASS>missense_variant,coding_sequence_variant</FXN_CLASS><VALIDATED>by-frequency,by-alfa,by-cluster</VALIDATED><DOCSUM>HGVS=NC_000019.10:g.44909057T&gt;A,NC_000019.9:g.45412314T&gt;A,NG_007084.2:g.8276T&gt;A,NM_000041.4:c.761T&gt;A,NM_000041.3:c.761T&gt;A,NM_000041.2:c.761T&gt;A,NM_001302688.2:c.839T&gt;A,NM_001302688.1:c.839T&gt;A,NM_001302690.2:c.761T&gt;A,NM_001302690.1:c.761T&gt;A,NM_001302691.2:c.761T&gt;A,NM_001302691.1:c.761T&gt;A,NM_001302689.2:c.761T&gt;A,NM_001302689.1:c.761T&gt;A,NP_000032.1:p.Val254Glu,NP_001289617.1:p.Val280Glu,NP_001289619.1:p.Val254Glu,NP_001289620.1:p.Val254Glu,NP_001289618.1:p.Val254Glu|SEQ=[T/A]|LEN=1|GENE=APOE:348</DOCSUM><TAX_ID>9606</TAX_ID><ORIG_BUILD>137</ORIG_BUILD><UPD_BUILD>156</UPD_BUILD><CREATEDATE>2012/05/04 13:09</CREATEDATE><UPDATEDATE>2022/10/13 16:55</UPDATEDATE><SS>491549551,646226060,713519655,994339190,1693665440,1698394133,2226012265,2744090652,2750218861,2963330156,2985144684,3017540732,3625742866,3653917431,3702845776,3825301413,5076165963,5307395595,5613191151,5847846247,5979547887</SS><ALLELE>W</ALLELE><SNP_CLASS>snv</SNP_CLASS><CHRPOS>19:44909057</CHRPOS><CHRPOS_PREV_ASSM>19:45412314</CHRPOS_PREV_ASSM><TEXT/><SNP_ID_SORT>0199768005</SNP_ID_SORT><CLINICAL_SORT>1</CLINICAL_SORT><CITED_SORT/><CHRPOS_SORT>0044909057</CHRPOS_SORT><MERGED_SORT>0</MERGED_SORT></DocumentSummary>\\n</ExchangeSet>'\n",
      "Going to download record 3 to 4\n",
      "b'<?xml version=\"1.0\" ?>\\n<ExchangeSet xmlns:xsi=\"https://www.w3.org/2001/XMLSchema-instance\" xmlns=\"https://www.ncbi.nlm.nih.gov/SNP/docsum\" xsi:schemaLocation=\"https://www.ncbi.nlm.nih.gov/SNP/docsum ftp://ftp.ncbi.nlm.nih.gov/snp/specs/docsum_eutils.xsd\" ><DocumentSummary uid=\"190853081\"><SNP_ID>190853081</SNP_ID><ALLELE_ORIGIN/><GLOBAL_MAFS><MAF><STUDY>1000Genomes</STUDY><FREQ>A=0.000781/4</FREQ></MAF><MAF><STUDY>ExAC</STUDY><FREQ>A=0.00043/13</FREQ></MAF><MAF><STUDY>GnomAD</STUDY><FREQ>A=0.000057/8</FREQ></MAF><MAF><STUDY>GnomAD_exomes</STUDY><FREQ>A=0.000203/39</FREQ></MAF><MAF><STUDY>KOREAN</STUDY><FREQ>A=0.006169/18</FREQ></MAF><MAF><STUDY>TOMMO</STUDY><FREQ>A=0.008639/145</FREQ></MAF><MAF><STUDY>TOPMED</STUDY><FREQ>A=0.000045/12</FREQ></MAF><MAF><STUDY>ALFA</STUDY><FREQ>A=0./0</FREQ></MAF></GLOBAL_MAFS><GLOBAL_POPULATION/><GLOBAL_SAMPLESIZE>0</GLOBAL_SAMPLESIZE><SUSPECTED/><CLINICAL_SIGNIFICANCE>pathogenic</CLINICAL_SIGNIFICANCE><GENES><GENE_E><NAME>APOE</NAME><GENE_ID>348</GENE_ID></GENE_E></GENES><ACC>NC_000019.10</ACC><CHR>19</CHR><HANDLE>NCBI-CURATED-RECORDS,GNOMAD,TOPMED,EVA,TOMMO_GENOMICS,ILLUMINA,EVA_EXAC,1000G_HIGH_COVERAGE,1000GENOMES,KRGDB</HANDLE><SPDI>NC_000019.10:44909082:G:A</SPDI><FXN_CLASS>missense_variant,coding_sequence_variant</FXN_CLASS><VALIDATED>by-frequency,by-alfa,by-cluster</VALIDATED><DOCSUM>HGVS=NC_000019.10:g.44909083G&gt;A,NC_000019.9:g.45412340G&gt;A,NG_007084.2:g.8302G&gt;A,NM_000041.4:c.787G&gt;A,NM_000041.3:c.787G&gt;A,NM_000041.2:c.787G&gt;A,NM_001302688.2:c.865G&gt;A,NM_001302688.1:c.865G&gt;A,NM_001302690.2:c.787G&gt;A,NM_001302690.1:c.787G&gt;A,NM_001302691.2:c.787G&gt;A,NM_001302691.1:c.787G&gt;A,NM_001302689.2:c.787G&gt;A,NM_001302689.1:c.787G&gt;A,NP_000032.1:p.Glu263Lys,NP_001289617.1:p.Glu289Lys,NP_001289619.1:p.Glu263Lys,NP_001289620.1:p.Glu263Lys,NP_001289618.1:p.Glu263Lys|SEQ=[G/A]|LEN=1|GENE=APOE:348</DOCSUM><TAX_ID>9606</TAX_ID><ORIG_BUILD>135</ORIG_BUILD><UPD_BUILD>156</UPD_BUILD><CREATEDATE>2011/09/17 04:22</CREATEDATE><UPDATEDATE>2022/10/13 16:55</UPDATEDATE><SS>465616514,537712922,1363326198,1693665447,1959869699,2744090661,2750218865,2963330160,3021911619,3625742871,3652337725,3725733607,3756111790,3938432709,5076165973,5227950609,5236962742,5237597852,5307395598,5435087084,5512463557,5613191154,5786644812,5847846249,5928256291,5953885395,5979547889</SS><ALLELE>R</ALLELE><SNP_CLASS>snv</SNP_CLASS><CHRPOS>19:44909083</CHRPOS><CHRPOS_PREV_ASSM>19:45412340</CHRPOS_PREV_ASSM><TEXT/><SNP_ID_SORT>0190853081</SNP_ID_SORT><CLINICAL_SORT>1</CLINICAL_SORT><CITED_SORT/><CHRPOS_SORT>0044909083</CHRPOS_SORT><MERGED_SORT>0</MERGED_SORT></DocumentSummary>\\n<DocumentSummary uid=\"140808909\"><SNP_ID>140808909</SNP_ID><ALLELE_ORIGIN/><GLOBAL_MAFS><MAF><STUDY>1000Genomes</STUDY><FREQ>A=0.000781/4</FREQ></MAF><MAF><STUDY>ExAC</STUDY><FREQ>A=0.000447/13</FREQ></MAF><MAF><STUDY>GnomAD</STUDY><FREQ>A=0.000057/8</FREQ></MAF><MAF><STUDY>GnomAD_exomes</STUDY><FREQ>A=0.000205/39</FREQ></MAF><MAF><STUDY>KOREAN</STUDY><FREQ>A=0.005483/16</FREQ></MAF><MAF><STUDY>PAGE_STUDY</STUDY><FREQ>A=0.000751/59</FREQ></MAF><MAF><STUDY>TOMMO</STUDY><FREQ>A=0.008639/145</FREQ></MAF><MAF><STUDY>TOPMED</STUDY><FREQ>A=0.000045/12</FREQ></MAF><MAF><STUDY>ALFA</STUDY><FREQ>A=0.000187/15</FREQ></MAF></GLOBAL_MAFS><GLOBAL_POPULATION/><GLOBAL_SAMPLESIZE>0</GLOBAL_SAMPLESIZE><SUSPECTED/><CLINICAL_SIGNIFICANCE>pathogenic</CLINICAL_SIGNIFICANCE><GENES><GENE_E><NAME>APOE</NAME><GENE_ID>348</GENE_ID></GENE_E></GENES><ACC>NC_000019.10</ACC><CHR>19</CHR><HANDLE>TOMMO_GENOMICS,KRGDB,PAGE_CC,TOPMED,1000G_HIGH_COVERAGE,ILLUMINA,NCBI-CURATED-RECORDS,1000GENOMES,GNOMAD,EVA_EXAC,EVA</HANDLE><SPDI>NC_000019.10:44909079:G:A,NC_000019.10:44909079:G:T</SPDI><FXN_CLASS>missense_variant,coding_sequence_variant,stop_gained</FXN_CLASS><VALIDATED>by-frequency,by-alfa,by-cluster</VALIDATED><DOCSUM>HGVS=NC_000019.10:g.44909080G&gt;A,NC_000019.10:g.44909080G&gt;T,NC_000019.9:g.45412337G&gt;A,NC_000019.9:g.45412337G&gt;T,NG_007084.2:g.8299G&gt;A,NG_007084.2:g.8299G&gt;T,NM_000041.4:c.784G&gt;A,NM_000041.4:c.784G&gt;T,NM_000041.3:c.784G&gt;A,NM_000041.3:c.784G&gt;T,NM_000041.2:c.784G&gt;A,NM_000041.2:c.784G&gt;T,NM_001302688.2:c.862G&gt;A,NM_001302688.2:c.862G&gt;T,NM_001302688.1:c.862G&gt;A,NM_001302688.1:c.862G&gt;T,NM_001302690.2:c.784G&gt;A,NM_001302690.2:c.784G&gt;T,NM_001302690.1:c.784G&gt;A,NM_001302690.1:c.784G&gt;T,NM_001302691.2:c.784G&gt;A,NM_001302691.2:c.784G&gt;T,NM_001302691.1:c.784G&gt;A,NM_001302691.1:c.784G&gt;T,NM_001302689.2:c.784G&gt;A,NM_001302689.2:c.784G&gt;T,NM_001302689.1:c.784G&gt;A,NM_001302689.1:c.784G&gt;T,NP_000032.1:p.Glu262Lys,NP_000032.1:p.Glu262Ter,NP_001289617.1:p.Glu288Lys,NP_001289617.1:p.Glu288Ter,NP_001289619.1:p.Glu262Lys,NP_001289619.1:p.Glu262Ter,NP_001289620.1:p.Glu262Lys,NP_001289620.1:p.Glu262Ter,NP_001289618.1:p.Glu262Lys,NP_001289618.1:p.Glu262Ter|SEQ=[G/A/T]|LEN=1|GENE=APOE:348</DOCSUM><TAX_ID>9606</TAX_ID><ORIG_BUILD>134</ORIG_BUILD><UPD_BUILD>156</UPD_BUILD><CREATEDATE>2011/05/09 23:43</CREATEDATE><UPDATEDATE>2022/10/13 16:55</UPDATEDATE><SS>340491284,537712921,1363326196,1693665445,1959869698,2095085225,2744090658,2750218864,2963330159,3021911617,3625742870,3652337722,3652337723,3725733605,3756111789,3772017001,3938432708,5076165971,5227950608,5236962741,5237597851,5307395596,5316121069,5435087082,5512463556,5613191152,5786644811,5847846248,5928256289,5953885394,5979547888</SS><ALLELE>D</ALLELE><SNP_CLASS>snv</SNP_CLASS><CHRPOS>19:44909080</CHRPOS><CHRPOS_PREV_ASSM>19:45412337</CHRPOS_PREV_ASSM><TEXT/><SNP_ID_SORT>0140808909</SNP_ID_SORT><CLINICAL_SORT>1</CLINICAL_SORT><CITED_SORT/><CHRPOS_SORT>0044909080</CHRPOS_SORT><MERGED_SORT>0</MERGED_SORT></DocumentSummary>\\n</ExchangeSet>'\n",
      "Going to download record 5 to 6\n",
      "b'<?xml version=\"1.0\" ?>\\n<ExchangeSet xmlns:xsi=\"https://www.w3.org/2001/XMLSchema-instance\" xmlns=\"https://www.ncbi.nlm.nih.gov/SNP/docsum\" xsi:schemaLocation=\"https://www.ncbi.nlm.nih.gov/SNP/docsum ftp://ftp.ncbi.nlm.nih.gov/snp/specs/docsum_eutils.xsd\" ><DocumentSummary uid=\"121918393\"><SNP_ID>121918393</SNP_ID><ALLELE_ORIGIN/><GLOBAL_MAFS><MAF><STUDY>1000Genomes</STUDY><FREQ>A=0.00031/2</FREQ></MAF><MAF><STUDY>ExAC</STUDY><FREQ>T=0.00013/2</FREQ></MAF><MAF><STUDY>SGDP_PRJ</STUDY><FREQ>C=0.5/1</FREQ></MAF><MAF><STUDY>ALFA</STUDY><FREQ>T=0./0</FREQ></MAF></GLOBAL_MAFS><GLOBAL_POPULATION/><GLOBAL_SAMPLESIZE>0</GLOBAL_SAMPLESIZE><SUSPECTED/><CLINICAL_SIGNIFICANCE>pathogenic,protective</CLINICAL_SIGNIFICANCE><GENES><GENE_E><NAME>APOE</NAME><GENE_ID>348</GENE_ID></GENE_E></GENES><ACC>NC_000019.10</ACC><CHR>19</CHR><HANDLE>OMIM-CURATED-RECORDS,EVA_DECODE,TOPMED,SGDP_PRJ,1000G_HIGH_COVERAGE,ILLUMINA,HUMAN_LONGEVITY,GNOMAD,EVA_EXAC,EVA</HANDLE><SPDI>NC_000019.10:44908755:C:A,NC_000019.10:44908755:C:T</SPDI><FXN_CLASS>missense_variant,coding_sequence_variant</FXN_CLASS><VALIDATED>by-frequency,by-alfa,by-cluster</VALIDATED><DOCSUM>HGVS=NC_000019.10:g.44908756C&gt;A,NC_000019.10:g.44908756C&gt;T,NC_000019.9:g.45412013C&gt;A,NC_000019.9:g.45412013C&gt;T,NG_007084.2:g.7975C&gt;A,NG_007084.2:g.7975C&gt;T,NM_000041.4:c.460C&gt;A,NM_000041.4:c.460C&gt;T,NM_000041.3:c.460C&gt;A,NM_000041.3:c.460C&gt;T,NM_000041.2:c.460C&gt;A,NM_000041.2:c.460C&gt;T,NM_001302688.2:c.538C&gt;A,NM_001302688.2:c.538C&gt;T,NM_001302688.1:c.538C&gt;A,NM_001302688.1:c.538C&gt;T,NM_001302690.2:c.460C&gt;A,NM_001302690.2:c.460C&gt;T,NM_001302690.1:c.460C&gt;A,NM_001302690.1:c.460C&gt;T,NM_001302691.2:c.460C&gt;A,NM_001302691.2:c.460C&gt;T,NM_001302691.1:c.460C&gt;A,NM_001302691.1:c.460C&gt;T,NM_001302689.2:c.460C&gt;A,NM_001302689.2:c.460C&gt;T,NM_001302689.1:c.460C&gt;A,NM_001302689.1:c.460C&gt;T,NP_000032.1:p.Arg154Ser,NP_000032.1:p.Arg154Cys,NP_001289617.1:p.Arg180Ser,NP_001289617.1:p.Arg180Cys,NP_001289619.1:p.Arg154Ser,NP_001289619.1:p.Arg154Cys,NP_001289620.1:p.Arg154Ser,NP_001289620.1:p.Arg154Cys,NP_001289618.1:p.Arg154Ser,NP_001289618.1:p.Arg154Cys|SEQ=[C/A/T]|LEN=1|GENE=APOE:348</DOCSUM><TAX_ID>9606</TAX_ID><ORIG_BUILD>133</ORIG_BUILD><UPD_BUILD>156</UPD_BUILD><CREATEDATE>2010/12/03 14:03</CREATEDATE><UPDATEDATE>2022/10/13 16:55</UPDATEDATE><SS>275517961,1693665415,1959869681,2226012254,2744090545,3021911599,3625742840,3625742841,3652337702,3702845770,3888305363,4331942413,4331942414,5076165847,5076165848,5307395585,5613191135,5799404797</SS><ALLELE>H</ALLELE><SNP_CLASS>snv</SNP_CLASS><CHRPOS>19:44908756</CHRPOS><CHRPOS_PREV_ASSM>19:45412013</CHRPOS_PREV_ASSM><TEXT/><SNP_ID_SORT>0121918393</SNP_ID_SORT><CLINICAL_SORT>1</CLINICAL_SORT><CITED_SORT/><CHRPOS_SORT>0044908756</CHRPOS_SORT><MERGED_SORT>0</MERGED_SORT></DocumentSummary>\\n<DocumentSummary uid=\"121918392\"><SNP_ID>121918392</SNP_ID><ALLELE_ORIGIN/><GLOBAL_MAFS><MAF><STUDY>1000Genomes</STUDY><FREQ>A=0.000156/1</FREQ></MAF><MAF><STUDY>ExAC</STUDY><FREQ>A=0.000082/8</FREQ></MAF><MAF><STUDY>GnomAD</STUDY><FREQ>A=0.000007/1</FREQ></MAF><MAF><STUDY>GnomAD_exomes</STUDY><FREQ>A=0.000036/9</FREQ></MAF><MAF><STUDY>KOREAN</STUDY><FREQ>A=0.001369/4</FREQ></MAF><MAF><STUDY>TOMMO</STUDY><FREQ>A=0.003787/63</FREQ></MAF><MAF><STUDY>TOPMED</STUDY><FREQ>A=0.000008/2</FREQ></MAF><MAF><STUDY>ALFA</STUDY><FREQ>A=0./0</FREQ></MAF></GLOBAL_MAFS><GLOBAL_POPULATION/><GLOBAL_SAMPLESIZE>0</GLOBAL_SAMPLESIZE><SUSPECTED/><CLINICAL_SIGNIFICANCE>pathogenic</CLINICAL_SIGNIFICANCE><GENES><GENE_E><NAME>APOE</NAME><GENE_ID>348</GENE_ID></GENE_E></GENES><ACC>NC_000019.10</ACC><CHR>19</CHR><HANDLE>TOMMO_GENOMICS,OMIM-CURATED-RECORDS,KRGDB,TOPMED,1000G_HIGH_COVERAGE,ILLUMINA,NCBI-CURATED-RECORDS,1000GENOMES,GNOMAD,EVA_EXAC,EVA</HANDLE><SPDI>NC_000019.10:44907776:G:A</SPDI><FXN_CLASS>missense_variant,coding_sequence_variant</FXN_CLASS><VALIDATED>by-frequency,by-alfa,by-cluster</VALIDATED><DOCSUM>HGVS=NC_000019.10:g.44907777G&gt;A,NC_000019.9:g.45411034G&gt;A,NG_007084.2:g.6996G&gt;A,NM_000041.4:c.61G&gt;A,NM_000041.3:c.61G&gt;A,NM_000041.2:c.61G&gt;A,NM_001302688.2:c.139G&gt;A,NM_001302688.1:c.139G&gt;A,NM_001302690.2:c.61G&gt;A,NM_001302690.1:c.61G&gt;A,NM_001302691.2:c.61G&gt;A,NM_001302691.1:c.61G&gt;A,NM_001302689.2:c.61G&gt;A,NM_001302689.1:c.61G&gt;A,NP_000032.1:p.Glu21Lys,NP_001289617.1:p.Glu47Lys,NP_001289619.1:p.Glu21Lys,NP_001289620.1:p.Glu21Lys,NP_001289618.1:p.Glu21Lys|SEQ=[G/A]|LEN=1|GENE=APOE:348</DOCSUM><TAX_ID>9606</TAX_ID><ORIG_BUILD>133</ORIG_BUILD><UPD_BUILD>156</UPD_BUILD><CREATEDATE>2010/12/03 14:03</CREATEDATE><UPDATEDATE>2022/10/13 16:55</UPDATEDATE><SS>275517960,340491279,537712920,1363326159,1693665308,1959869646,2095085216,2744090372,3021911538,3625742789,3652337634,3652337635,3653917423,3725733567,3756111779,3938432664,4331942272,5076165561,5227950562,5435087013,5613191097,5786644755,5928256247</SS><ALLELE>R</ALLELE><SNP_CLASS>snv</SNP_CLASS><CHRPOS>19:44907777</CHRPOS><CHRPOS_PREV_ASSM>19:45411034</CHRPOS_PREV_ASSM><TEXT/><SNP_ID_SORT>0121918392</SNP_ID_SORT><CLINICAL_SORT>1</CLINICAL_SORT><CITED_SORT/><CHRPOS_SORT>0044907777</CHRPOS_SORT><MERGED_SORT>0</MERGED_SORT></DocumentSummary>\\n</ExchangeSet>'\n",
      "Going to download record 7 to 8\n",
      "b'<?xml version=\"1.0\" ?>\\n<ExchangeSet xmlns:xsi=\"https://www.w3.org/2001/XMLSchema-instance\" xmlns=\"https://www.ncbi.nlm.nih.gov/SNP/docsum\" xsi:schemaLocation=\"https://www.ncbi.nlm.nih.gov/SNP/docsum ftp://ftp.ncbi.nlm.nih.gov/snp/specs/docsum_eutils.xsd\" ><DocumentSummary uid=\"769455\"><SNP_ID>769455</SNP_ID><ALLELE_ORIGIN/><GLOBAL_MAFS><MAF><STUDY>1000Genomes</STUDY><FREQ>T=0.007183/36</FREQ></MAF><MAF><STUDY>ExAC</STUDY><FREQ>T=0.002635/39</FREQ></MAF><MAF><STUDY>GnomAD</STUDY><FREQ>T=0.007012/983</FREQ></MAF><MAF><STUDY>GnomAD_exomes</STUDY><FREQ>T=0.001421/220</FREQ></MAF><MAF><STUDY>HapMap</STUDY><FREQ>T=0.025547/7</FREQ></MAF><MAF><STUDY>Qatari</STUDY><FREQ>T=0.009259/2</FREQ></MAF><MAF><STUDY>SGDP_PRJ</STUDY><FREQ>C=0.5/1</FREQ></MAF><MAF><STUDY>TOPMED</STUDY><FREQ>T=0.007847/2077</FREQ></MAF><MAF><STUDY>ALFA</STUDY><FREQ>T=0.001041/119</FREQ></MAF></GLOBAL_MAFS><GLOBAL_POPULATION/><GLOBAL_SAMPLESIZE>0</GLOBAL_SAMPLESIZE><SUSPECTED/><CLINICAL_SIGNIFICANCE>benign,conflicting-interpretations-of-pathogenicity,pathogenic,pathogenic-likely-pathogenic</CLINICAL_SIGNIFICANCE><GENES><GENE_E><NAME>APOE</NAME><GENE_ID>348</GENE_ID></GENE_E></GENES><ACC>NC_000019.10</ACC><CHR>19</CHR><HANDLE>WEILL_CORNELL_DGM,CSHL-HAPMAP,ENSEMBL,1000GENOMES,1000G_HIGH_COVERAGE,DEBNICK,KHV_HUMAN_GENOMES,EVA,OMICIA,EXOME_CHIP,TRAN_CS_UWATERLOO,SANFORD_IMAGENETICS,HUMAN_LONGEVITY,NHLBI-ESP,ABI,SGDP_PRJ,GNOMAD,TOPMED,NCBI-CURATED-RECORDS,ILLUMINA,EVA_DECODE,KRIBB_YJKIM,EVA_EXAC,AFFY,OMIM-CURATED-RECORDS,HUGCELL_USP,RSG_UW,SEATTLESEQ</HANDLE><SPDI>NC_000019.10:44908782:C:T</SPDI><FXN_CLASS>coding_sequence_variant,missense_variant</FXN_CLASS><VALIDATED>by-frequency,by-alfa,by-cluster</VALIDATED><DOCSUM>HGVS=NC_000019.10:g.44908783C&gt;T,NC_000019.9:g.45412040C&gt;T,NG_007084.2:g.8002C&gt;T,NM_000041.4:c.487C&gt;T,NM_000041.3:c.487C&gt;T,NM_000041.2:c.487C&gt;T,NM_001302688.2:c.565C&gt;T,NM_001302688.1:c.565C&gt;T,NM_001302690.2:c.487C&gt;T,NM_001302690.1:c.487C&gt;T,NM_001302691.2:c.487C&gt;T,NM_001302691.1:c.487C&gt;T,NM_001302689.2:c.487C&gt;T,NM_001302689.1:c.487C&gt;T,NP_000032.1:p.Arg163Cys,NP_001289617.1:p.Arg189Cys,NP_001289619.1:p.Arg163Cys,NP_001289620.1:p.Arg163Cys,NP_001289618.1:p.Arg163Cys|SEQ=[C/T]|LEN=1|GENE=APOE:348</DOCSUM><TAX_ID>9606</TAX_ID><ORIG_BUILD>86</ORIG_BUILD><UPD_BUILD>156</UPD_BUILD><CREATEDATE>2000/10/05 19:10</CREATEDATE><UPDATEDATE>2022/10/13 16:55</UPDATEDATE><SS>870164,17624004,40972757,75046407,107936538,119367435,159740186,161880957,174499350,228156554,244239671,244310559,275517962,410828264,482349006,483834110,491161752,491549549,537712915,713519651,781732084,783424857,1363326190,1693665418,1752288223,1917938920,1937835822,1946537657,1959869687,1959869688,2094804139,2095085223,2226012257,2744090560,2750218831,2963330126,2985144681,3021911604,3021911605,3625742842,3634738105,3638229876,3640445413,3641098340,3641394037,3643202422,3644726172,3652337707,3652337708,3652337709,3653917427,3702845771,3725733600,3744466129,3745038114,3772535198,3821359904,3825301409,3888305365,3986802273,4017826838,5076165856,5307395587,5314453656,5435087068,5499891499,5613191138,5662475603,5847846242,5928256283,5953885388</SS><ALLELE>Y</ALLELE><SNP_CLASS>snv</SNP_CLASS><CHRPOS>19:44908783</CHRPOS><CHRPOS_PREV_ASSM>19:45412040</CHRPOS_PREV_ASSM><TEXT/><SNP_ID_SORT>0000769455</SNP_ID_SORT><CLINICAL_SORT>1</CLINICAL_SORT><CITED_SORT/><CHRPOS_SORT>0044908783</CHRPOS_SORT><MERGED_SORT>0</MERGED_SORT></DocumentSummary>\\n<DocumentSummary uid=\"769452\"><SNP_ID>769452</SNP_ID><ALLELE_ORIGIN/><GLOBAL_MAFS><MAF><STUDY>1000Genomes</STUDY><FREQ>C=0.000937/5</FREQ></MAF><MAF><STUDY>ALSPAC</STUDY><FREQ>C=0.000519/2</FREQ></MAF><MAF><STUDY>Estonian</STUDY><FREQ>C=0.005804/26</FREQ></MAF><MAF><STUDY>ExAC</STUDY><FREQ>C=0.002423/288</FREQ></MAF><MAF><STUDY>FINRISK</STUDY><FREQ>C=0.009934/3</FREQ></MAF><MAF><STUDY>GnomAD</STUDY><FREQ>C=0.001989/279</FREQ></MAF><MAF><STUDY>GnomAD_exomes</STUDY><FREQ>C=0.002509/629</FREQ></MAF><MAF><STUDY>GoNL</STUDY><FREQ>C=0.001002/1</FREQ></MAF><MAF><STUDY>Korea1K</STUDY><FREQ>A=0.003821/7</FREQ></MAF><MAF><STUDY>NorthernSweden</STUDY><FREQ>C=0.008333/5</FREQ></MAF><MAF><STUDY>PAGE_STUDY</STUDY><FREQ>C=0.000356/28</FREQ></MAF><MAF><STUDY>PRJEB37766</STUDY><FREQ>C=0.000611/2</FREQ></MAF><MAF><STUDY>TOPMED</STUDY><FREQ>C=0.0013/344</FREQ></MAF><MAF><STUDY>TWINSUK</STUDY><FREQ>C=0.002157/8</FREQ></MAF><MAF><STUDY>ALFA</STUDY><FREQ>C=0.002363/520</FREQ></MAF></GLOBAL_MAFS><GLOBAL_POPULATION/><GLOBAL_SAMPLESIZE>0</GLOBAL_SAMPLESIZE><SUSPECTED/><CLINICAL_SIGNIFICANCE>uncertain-significance,conflicting-interpretations-of-pathogenicity,pathogenic,likely-pathogenic</CLINICAL_SIGNIFICANCE><GENES><GENE_E><NAME>APOE</NAME><GENE_ID>348</GENE_ID></GENE_E></GENES><ACC>NC_000019.10</ACC><CHR>19</CHR><HANDLE>SWEGEN,1000GENOMES,EVA-GONL,1000G_HIGH_COVERAGE,DEBNICK,EVA,SI_EXO,OMICIA,EVA_FINRISK,EXOME_CHIP,SANFORD_IMAGENETICS,PAGE_CC,HUMAN_LONGEVITY,NHLBI-ESP,CLINSEQ_SNP,EVA_UK10K_TWINSUK,GNOMAD,TOPMED,ACPOP,ILLUMINA,EGCUT_WGS,EVA_DECODE,EVA_EXAC,AFFY,EVA_UK10K_ALSPAC,HUGCELL_USP,KOGIC</HANDLE><SPDI>NC_000019.10:44907852:T:A,NC_000019.10:44907852:T:C</SPDI><FXN_CLASS>coding_sequence_variant,missense_variant</FXN_CLASS><VALIDATED>by-frequency,by-alfa,by-cluster</VALIDATED><DOCSUM>HGVS=NC_000019.10:g.44907853T&gt;A,NC_000019.10:g.44907853T&gt;C,NC_000019.9:g.45411110T&gt;A,NC_000019.9:g.45411110T&gt;C,NG_007084.2:g.7072T&gt;A,NG_007084.2:g.7072T&gt;C,NM_000041.4:c.137T&gt;A,NM_000041.4:c.137T&gt;C,NM_000041.3:c.137T&gt;A,NM_000041.3:c.137T&gt;C,NM_000041.2:c.137T&gt;A,NM_000041.2:c.137T&gt;C,NM_001302688.2:c.215T&gt;A,NM_001302688.2:c.215T&gt;C,NM_001302688.1:c.215T&gt;A,NM_001302688.1:c.215T&gt;C,NM_001302690.2:c.137T&gt;A,NM_001302690.2:c.137T&gt;C,NM_001302690.1:c.137T&gt;A,NM_001302690.1:c.137T&gt;C,NM_001302691.2:c.137T&gt;A,NM_001302691.2:c.137T&gt;C,NM_001302691.1:c.137T&gt;A,NM_001302691.1:c.137T&gt;C,NM_001302689.2:c.137T&gt;A,NM_001302689.2:c.137T&gt;C,NM_001302689.1:c.137T&gt;A,NM_001302689.1:c.137T&gt;C,NP_000032.1:p.Leu46Gln,NP_000032.1:p.Leu46Pro,NP_001289617.1:p.Leu72Gln,NP_001289617.1:p.Leu72Pro,NP_001289619.1:p.Leu46Gln,NP_001289619.1:p.Leu46Pro,NP_001289620.1:p.Leu46Gln,NP_001289620.1:p.Leu46Pro,NP_001289618.1:p.Leu46Gln,NP_001289618.1:p.Leu46Pro|SEQ=[T/A/C]|LEN=1|GENE=APOE:348</DOCSUM><TAX_ID>9606</TAX_ID><ORIG_BUILD>86</ORIG_BUILD><UPD_BUILD>156</UPD_BUILD><CREATEDATE>2000/10/05 19:10</CREATEDATE><UPDATEDATE>2022/10/13 16:55</UPDATEDATE><SS>870160,76889657,244239684,342502233,489170611,491549547,491769735,780747096,783424855,994339185,1363326161,1584118335,1638043648,1681037681,1693665327,1698394127,1752288222,1917938919,1946537656,1959869650,2226012197,2710884164,2744090395,2750218799,2963330067,2985144679,3017540723,3021911542,3625742800,3627941612,3634738104,3640445412,3644726171,3652337642,3653917424,3684286595,3725733571,3743072609,3744466119,3745038113,3772016990,3772535197,3825301393,3825938282,3981436455,3984450567,3986802269,5076165583,5237672907,5307395572,5435087020,5499891488,5613191100,5624429770,5662475594,5840606764,5847846236,5928256252,5953885383,5979547881</SS><ALLELE>H</ALLELE><SNP_CLASS>snv</SNP_CLASS><CHRPOS>19:44907853</CHRPOS><CHRPOS_PREV_ASSM>19:45411110</CHRPOS_PREV_ASSM><TEXT/><SNP_ID_SORT>0000769452</SNP_ID_SORT><CLINICAL_SORT>1</CLINICAL_SORT><CITED_SORT/><CHRPOS_SORT>0044907853</CHRPOS_SORT><MERGED_SORT>0</MERGED_SORT></DocumentSummary>\\n</ExchangeSet>'\n"
     ]
    }
   ],
   "source": [
    "# sample codes adopted with modifications from http://biopython.org/DIST/docs/tutorial/Tutorial.html#htoc139.\n",
    "fetch_count = 0\n",
    "for start in range(0, total_count, retmax):\n",
    "    end = min(total_count, start+retmax)\n",
    "    print(\"Going to download record %i to %i\" % (start+1, end))\n",
    "    attempt = 0\n",
    "    #fetch_count += 1\n",
    "    while (attempt < 3):\n",
    "        attempt += 1\n",
    "        try:\n",
    "            fetch_handle = Entrez.efetch(db=\"snp\",\n",
    "                                         retmode=\"xml\",\n",
    "                                         retstart=start,\n",
    "                                         retmax=retmax,\n",
    "                                         webenv=webenv,\n",
    "                                         query_key=query_key)\n",
    "            break  # Добавьте это прерывание цикла после успешного выполнения запроса\n",
    "        except HTTPError as err:\n",
    "            if 500 <= err.code <= 599:\n",
    "                print(\"Received error from server %s\" % err)\n",
    "                print(\"Attempt %i of 3\" % attempt)\n",
    "                time.sleep(15)\n",
    "            else:\n",
    "                raise\n",
    "    if (fetch_handle):\n",
    "        #print(fetch_handle)\n",
    "        data = fetch_handle.read()\n",
    "        print(data)\n",
    "        fetch_handle.close()"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of RSIDs: 8\n"
     ]
    }
   ],
   "source": [
    "# Печать количества RSID\n",
    "print(\"Total number of RSIDs:\", len(rslist))"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "\n",
    "\n"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}